lib/puppet/transaction/report.rb

```diff
@@ -386,6 +394,101 @@ def self.default_format
 
   private
 
+  # Using the last applied values from our persistence store, and the current
+  # report data, iterate across all events finding any corrective_change that
+  # might exist and marking the event, resource and report appropriately.
+  def process_resource_inferences
+    persistence = Puppet::Transaction::Persistence.new
```

Comment:
I still don't like that we're tying together the loading/saving of persistence and doing the corrective change calculations in the same function; I would rather have them decoupled to separate the responsibilities and make it easier to test things.